test/functions.jl

```diff
@@ -296,6 +296,12 @@ using Statistics
         @test isapprox(nda, nda4; atol=2π)
     end
 
+    @testset "deleteat!" begin
+        ndv = NamedDimsArray{(:a,)}([1, 2, 3, 4, 5, 6.0])
+        @test deleteat!(ndv, 1) == NamedDimsArray{(:a,)}([2, 3, 4, 5, 6.0])
+        @test deleteat!(ndv, 1:2) == NamedDimsArray{(:a,)}([4, 5, 6.0])
+        @test deleteat!(ndv, (1, 3)) == NamedDimsArray{(:a,)}([5.0])
```

Comment:
Should we test that it mutates? I.e.

```suggestion
        deleteat!(ndv, 1) 
        @test ndv == NamedDimsArray{(:a,)}([2, 3, 4, 5, 6.0])
        deleteat!(ndv, 1:2) 
        @test ndv == NamedDimsArray{(:a,)}([4, 5, 6.0])
        deleteat!(ndv, (1, 3))
        @test ndv == NamedDimsArray{(:a,)}([5.0])
```

Reply:
We are implicitly testing for mutation already - otherwise the array would not shrink as we go. Do you think being explicit about testing mutation is worth the extra lines?

Reply:
Another option
```suggestion
        @test ndv == deleteat!(ndv, 1) == NamedDimsArray{(:a,)}([2, 3, 4, 5, 6.0])
        @test ndv == deleteat!(ndv, 1:2) == NamedDimsArray{(:a,)}([4, 5, 6.0])
        @test ndv == deleteat!(ndv, (1, 3)) == NamedDimsArray{(:a,)}([5.0])
```